File: 이건희/level1/명예의 전당 (1).py
def solution(k, score):
    result = []
    award = []
    for i in range(len(score)):
        award.append(score[i])
        award.sort(reverse = True)
        if len(award) < k:
            result.append(min(award))
        else:
            result.append(award[k - 1])

    return result

File: 이훈일/Programmers/Level1/명예의 전당 (1).java
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

class Solution {
    public int[] solution(int k, int[] score) {
        List<Integer> best = new ArrayList<>();
        int[] result = new int[score.length];
        for(int i = 0; i < score.length; i++) {
            best.add(score[i]);
            Collections.sort(best);
            if(best.size() == k+1) {
                best.remove(0);
            }
            result[i] = best.get(0);
        }

        return result;
    }
}
